### PR TITLE
More API module splitting

### PR DESCRIPTION
**Pull request by dcoutts**
Split almost all the remaining things out of Cardano.Api.Typed
so they are lots of smaller modules. This nearly completes the
split of the original monolithic module. The only remaining part
is the IPC with the node, which will be replaces shortly.

### CODE REVIEW

**Comment by dcoutts** [comment]
Note that this builds on top of #2093. That one is a single patch. It can either be merged on its own or as part of this PR. It's not directly related to the module splitting however, so it is perhaps neater to keep it separate.

File: cardano-api/src/Cardano/Api/Typed.hs [hunk 1]
```diff
@@ -85,17 +86,42 @@ module Cardano.Api.Typed (
     StakeKey,
     StakeExtendedKey,
 
+    -- * Currency values
+    -- ** Ada \/ Lovelace
+    Lovelace(..),
+
+    -- ** Multi-asset values
```
Comment (Jimbo4350) [inline cardano-api/src/Cardano/Api/Typed.hs:85]: Aren't we supposed to be slowly getting rid of `Typed.hs`?

**Review by Jimbo4350 — APPROVED** [review]
LGTM!

**Review by intricate — APPROVED** [review]

**Pull request merged**